### PR TITLE
Hide left & right navs on mdx pages with frontmatter

### PR DESCRIPTION
**Pull request by tabathadelane**
if `hideNavs: true` frontmatter is added to a doc, the page layout changes. The left nav will collapse, and the right rail will be removed. 

Some docs pages are in experiments as JS, so there is also an added check for those pathnames.

For preview purposes, there is a temporary frontmatter added to 
 - [docs/new-relic-solutions/get-started/quick-launch-guide/](https://docswebsitedevelop-tabathahidebothnavs.gatsbyjs.io/docs/new-relic-solutions/get-started/quick-launch-guide/)
 - [docs/apm/new-relic-apm/getting-started/introduction-apm/](https://docswebsitedevelop-tabathahidebothnavs.gatsbyjs.io/introduction-apm/) (JS page)
 - [docs/apm/new-relic-apm/getting-started/apm-agent-data-security/](https://docswebsitedevelop-tabathahidebothnavs.gatsbyjs.io/docs/apm/new-relic-apm/getting-started/apm-agent-data-security/)

TODO:
 - [x] remove the frontmatter from mdx pages

### CODE REVIEW

File: src/layouts/MainLayout.js [hunk 1]
```diff
@@ -41,9 +42,21 @@ const MainLayout = ({ children, pageContext, sidebarOpen = true }) => {
...
+   * [VSU] some docs pages are being designed as JS for faster experimenting
+   * and will never have the frontmatter property
+   * Using regex for check to account for paths with and without trailing slash
+   */
+  const docsAsJS = ['/introduction-apm'];
+  docsAsJS.forEach((docUrl) => {
+    if (location.pathname.match(docUrl) && !hideNavs) {
+      hideNavs = true;
+    }
+  });
```
Comment (sunnyzanchi) [inline src/layouts/MainLayout.js:55]: how about this?
```suggestion
  const isJSDoc = docsAsJS.some((docUrl) => docUrl.test(location.pathname));
  hideNavs ||= isJSDoc
```

`String#match` would work here, but it returns either an array of matches or `null`, whereas `RegExp#test` returns a boolean, which feels more appropriate here

File: src/layouts/MainLayout.js [hunk 2]
```diff
@@ -41,9 +42,21 @@ const MainLayout = ({ children, pageContext, sidebarOpen = true }) => {
     }
   };
 
+  /*
+   * [VSU] some docs pages are being designed as JS for faster experimenting
+   * and will never have the frontmatter property
+   * Using regex for check to account for paths with and without trailing slash
+   */
+  const docsAsJS = ['/introduction-apm'];
```
Comment (sunnyzanchi) [inline src/layouts/MainLayout.js:50]: did you mean this?
```suggestion
  const docsAsJS = [/introduction-apm/];
```
`String#match` will automatically try to convert its argument to a regex, so the string `'/introduction-apm'` is converted to `/\/introduction-apm/`

File: src/layouts/MainLayout.js [hunk 3]
```diff
@@ -41,9 +42,21 @@ const MainLayout = ({ children, pageContext, sidebarOpen = true }) => {
...
+  docsAsJS.forEach((docUrl) => {
+    if (location.pathname.match(docUrl) && !hideNavs) {
+      hideNavs = true;
+    }
+  });
+
   useEffect(() => {
     setIsMobileNavOpen(false);
-    setSidebar(sidebarOpen);
+    setSidebar(hideNavs ? !hideNavs : sidebarOpen);
```
Comment (sunnyzanchi) [inline src/layouts/MainLayout.js:59]: nit: these are technically equivalent but i think it'd be more clear to just use `false`

```suggestion
    setSidebar(hideNavs ? false : sidebarOpen);
```

**Review by sunnyzanchi — CHANGES_REQUESTED** [review]
looks good, just left a couple comments

File: src/layouts/MainLayout.js (hunk 3 above)
Comment (tabathadelane) [reply src/layouts/MainLayout.js:59]: Classic overthinking it.

File: src/layouts/MainLayout.js (hunk 2 above)
Comment (tabathadelane) [reply src/layouts/MainLayout.js:50]: Okay. I was wondering about these Regex nuances, but it was behaving as expected, so I just left it to the will of the PR gods. I'll make an update.

**Review by sunnyzanchi — APPROVED** [review]
✅

**Review by LizBaker — DISMISSED** [review]

**Comment by tabathadelane** [comment]
I'm going to remove the frontmatter from testing which will dismiss the reviews. I'll just do an override once the checks finish. 👍

**Review by sunnyzanchi — APPROVED** [review]

**Pull request merged**